filebeat/beater/filebeat.go

```diff
@@ -92,15 +99,23 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	// Stopping spooler will flush items
 	defer spooler.Stop()
 
-	err = crawler.Start(registrar.GetStates())
+	err = crawler.Start(registrar.GetStates(), *once)
 	if err != nil {
 		return err
 	}
 	// Stop crawler -> stop prospectors -> stop harvesters
 	defer crawler.Stop()
 
-	// Blocks progressing. As soon as channel is closed, all defer statements come into play
-	<-fb.done
+	if *once {
+		logp.Debug("filebeat", "Running filebeat once. Waiting for completion ...")
+		spooler.Flush()
+		publisher.Publish()
```

Comment:
@urso What I'm trying to do here is to make sure that all events which are in the queue are sent. Is there a better way to wait / check the publisher if all events were sent?